src/Microsoft.OpenApi/Extensions/OpenApiSerializableExtensions.cs

```diff
@@ -15,6 +15,8 @@ namespace Microsoft.OpenApi.Extensions
     /// </summary>
     public static class OpenApiSerializableExtensions
     {
+        public delegate void SerializeDelegate(IOpenApiWriter writer, IOpenApiSerializable element);
```

Comment:
can you provide some context on the introduction of the delegate here?

Reply:
This was the only place I could think of that would expose the delegate to all implementers of the IOpenApiSerializable interface.

Reply:
instead of using a delegate, why not using an Action?

```CSharp
public void foo(Action<IOpenApiSerializable, IOpenApiWriter> callback)
```

Sorry if I wasn't clear in my earlier comments, I never meant to expose the callback outside of the library, I meant that as an alternative to having to copy/paste the serialize method implementation.

Reply:
Noted.